search-service/src/main/kotlin/com/egm/stellio/search/temporal/util/TemporalQueryUtils.kt

```diff
@@ -206,3 +189,41 @@ fun buildTimerelAndTime(
     } else {
         "'timerel' and 'time' must be used in conjunction".left()
     }
+
+private fun extractTemporalRepresentation(
+    queryParams: MultiValueMap<String, String>
+): Either<APIException, TemporalRepresentation> = either {
+    val optionsParam = Optional.ofNullable(queryParams.getFirst(QueryParameter.OPTIONS.key))
+    val formatParam = queryParams.getFirst(QueryParameter.FORMAT.key)
+    if (formatParam != null) {
+        return TemporalRepresentation.fromString(formatParam)
+    } else {
+        return if (!optionsParam.isEmpty) {
+            val hasTemporal = hasValueInOptionsParam(optionsParam, OptionsValue.TEMPORAL_VALUES).bind()
+            val hasAggregated = hasValueInOptionsParam(optionsParam, OptionsValue.AGGREGATED_VALUES).bind()
+            when {
+                hasTemporal && hasAggregated ->
+                    BadRequestDataException(
```

Comment:
it should be an `InvalidRequest` (as when the `options` contains unsupported values